templates/flavors/system-assigned-identity/patches/system-assigned-identity.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
```

Comment:
Do you need this patch file? It seems like `../base` + `./system-assigned-identity.yaml` together have all the resources needed.

Reply:
its removing the clientid and secret from the `KubeadmConfigTemplate`

Reply:
Agreed, but you also have that same resource described in `./templates/flavors/system-assigned-identity/system-assigned-identitiy.yaml`

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/565/files#diff-02c4b6da812bc0b2eb4eed8c28e6fc04R45

Reply:
its adding `identity: SystemAssigned` to the AzureMachineTemplate | ${CLUSTER_NAME}-control-plane